### ISSUE TITLE
Monitoring console pod keeps restarting

### ISSUE THREAD

**Issue by adrianmo** [splunk/splunk-operator#330]
After going though the [Multisite Deployment](https://github.com/splunk/splunk-operator/blob/develop/docs/MultisiteExamples.md) docs to deploy a zone-aware cluster, we updated the affinity spec of the ClusterMaster and the IndexerCluster. Both the ClusterMaster and IndexerCluster pods were scheduled to the expected nodes, however the Monitoring Console pod kept restarting in an infinite loop.

ClusterMaster spec:

```yaml
apiVersion: enterprise.splunk.com/v1
kind: ClusterMaster
metadata:
  name: cm
  namespace: splunk
  finalizers:
  - enterprise.splunk.com/delete-pvc
  defaults: |-
    splunk:
      site: site1
      multisite_master: localhost
      all_sites: site1,site2,site3
      multisite_replication_factor_origin: 1
      multisite_replication_factor_total: 3
      multisite_search_factor_origin: 1
      multisite_search_factor_total: 3
      idxc:
        search_factor: 3
        replication_factor: 3
  licenseMasterRef:
    name: lm
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: agentpool
            operator: In
            values:
            - splunk
          - key: topology.kubernetes.io/zone
            operator: In
            values:
            - west-europe-1
```

After some investigation we found out that the Monitoring Console StatefulSet, which is driven by the specs of the ClusterMaster, LicenseMaster, and the SearchHeadCluster, was being constantly updated with contradicting affinity definitions which in turn caused the underlying pod to be recreated continually.

When the Monitoring Console StatefulSet was updated from the ClusterMaster reconcile loop, the zone-aware affinity was added; then, when it was updated from the LicenseMaster or the SearchHeadCluster loops, the affinity was back to the default value.

Removing the affinity from the ClusterMaster, or adding the same zone-aware affinity to the LicenseMaster and the SearchHeadCluster made the Monitoring Console pod stable. 

The question that comes to mind is: should the Monitoring Console affinity be inherited from the other CRs, or the Monitoring Console affinity should be completely independent? If so, maybe it would deserve its own CR...

**Comment by smohan-splunk** [comment]
@adrianmo Thanks for reporting this issue to us. We are aware of this issue (MC restarting when spec params don't match between other CRs). In the past we tried workaround of hard coding some of the resource params such as CPU & Memory for MC, but you are absolutely right - the long term solution will be to create a CR for MC. 

Our teams are currently investing time on requirements for MC's CR, and I hope we will have this included in the near future. Please stay tuned!

**Comment by adrianmo** [comment]
@smohan-splunk Thanks for the quick response and acknowledging the issue. For the time being, and despite not being able to achieve our desired configuration, we will apply one of the patches I mentioned above. We will keep an eye on upcoming updates!

**Comment by onlineid** [comment]
@adrianmo Hi, I may have same problem. Monitoring Console keeps restarting after I customized the affinity rules on Cluster Master and Search Head Cluster. Should I apply same affinity rules to all statefulset? Thanks.

**Comment by kashok-splunk** [comment]
Hi @adrianmo the above issue has been resolved after the Splunk Operator 1.1 release, which contains separate CRD for MC.
